src/compiler/checker.ts

```diff
@@ -1458,9 +1458,8 @@ namespace ts {
                     case SyntaxKind.JSDocTypedefTag:
                     case SyntaxKind.JSDocCallbackTag:
                         // js type aliases do not resolve names from their host, so skip past it
-                        lastLocation = location;
-                        location = getJSDocHost(location).parent;
-                        continue;
+                        location = getJSDocHost(location);

```

Comment:
Note: we still go to `.parent` a few lines down in the usual way, so the only effect is on `lastLocation`.